homeassistant/components/notify/matrix.py

```diff
@@ -0,0 +1,170 @@
+"""
+Matrix notification service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/notify.matrix/
+"""
+import logging
+import json
+import os
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.notify import (
+    ATTR_TARGET, PLATFORM_SCHEMA, BaseNotificationService)
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD, CONF_VERIFY_SSL
+
+REQUIREMENTS = ['matrix-client==0.0.5']
+
+SESSION_FILE = 'matrix.conf'
+AUTH_TOKENS = None
+
+CONF_HOMESERVER = 'homeserver'
+CONF_DEFAULT_ROOM = 'default_room'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOMESERVER): cv.url,
+    vol.Optional(CONF_VERIFY_SSL, default=True): cv.boolean,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_DEFAULT_ROOM): cv.string,
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_service(hass, config):
+    """Get the Matrix notification service."""
+    load_token(hass.config.path(SESSION_FILE))
```

Comment:
An alternative could be to have all instances fall under one banner but that might cause more confusion to the user.

Another alternative is to load it once and store it in a global value.